### ISSUE TITLE
[BUG] Can't make an account with an edu address

### ISSUE THREAD

**Issue by Felienne** [hedyorg/hedy#4726]
I guess our email verification is a bit too strict!

<img width="805" alt="image" src="https://github.com/hedyorg/hedy/assets/1003685/c39d7da6-ab12-4f4d-9003-1f6f84c96769">

This is where the validation happens:
https://github.com/hedyorg/hedy/blob/4c202a1466b41d09883d4fa14d64f8718371250b/utils.py#L179

No clue what this exactly does (it is very old code without any explanation, sadly)

Maybe we can use this:
https://emailregex.com/

It is also unreadable but at least it has documentation that we can put in a comment :)

**Comment by Felienne** [comment]
In #4730 I have hotpatched the specific issue with .education, but I would still like to use something more robust/compliant with the spec, so I will leave this open.

**Comment by jpelay** [comment]
The email regex grammar is notoriously hard to parse with regex! I suggest we use a library specifically designed for this purpose.

**Comment by Felienne** [comment]
> The email regex grammar is notoriously hard to parse with regex! I suggest we use a library specifically designed for this purpose.

That sounds like a better plan that hacking away at regexes! The immediate problem of the teacher reporting this is fixed for now, but surely we want a better solution here!
